daras_ai_v2/azure_doc_extract.py

```diff
@@ -54,7 +56,14 @@ def azure_form_recognizer_models() -> dict[str, str]:
     return {value["modelId"]: value["description"] for value in r.json()["value"]}
 
 
+def azure_should_retry(e: Exception) -> bool:
+    return isinstance(e, requests.exceptions.RequestException) and (
+        e.response.status_code in (408, 429)  # Retry for rate limits or timeouts
+    )
```

Comment:
response can be `None`

<img width="748" alt="image" src="https://github.com/user-attachments/assets/aafa0b29-dac4-4c61-bb9b-b63499157c9f">

please also retry for standard HTTP codes - 
https://github.com/GooeyAI/aifail/blob/577487ad5ccbf3d70d4610a6b7d09e45a4951a73/aifail/aifail.py#L71 (or just reuse `http_should_retry` here)